docs: add dedicated package for code snippets from contracts section (#895)
FuelLabs/fuels-ts: apps/docs/src/guide/contracts/calls-with-different-wallets.md

# Making Calls with Different Wallets or Providers

> This guide demonstrates how to make contract calls using different wallets or providers by modifying a contract instance's `account` and `provider` properties.

Reviewer: ```suggestion
This guide demonstrates how to make contract calls using different wallets and providers by passing either an `Account` or a `Provider` to the contract on instantiation.
```
My understanding is that you should provide one of the above to a contract instead of both?

https://github.com/FuelLabs/fuels-ts/blob/d0eb7a39d2d5cd59cc45fede3826a327f158d5ea/packages/program/src/contract.ts#L20-L24

Should we be recommending that a consumer can alter the provider after class instantiation? As the logic inside the constructor looks quite important. Should the `account` and `provider` properties be `readonly`?

Consider the following scenario:
1. Contract is instantiated with a `Provider`
2. Consumer alters the `account` property where the passed `Account` instance has a `provider` property that is `!==` to the `Provider` instance that was passed to the contract on instantiation
3. The `getBalance()` method is called and the balance is retrieved by the initial `Provider` as we have not updated to the account provider, as this functionality only exists in the class constructor
4. This could be success or fail, dependent on the expected functionality from our consumer, which 'Provider' were they expecting to be used?

Early days for me so may be misunderstanding so apologies. Out of scope of this PR, but it has got me thinking 🤔

Author: I think you raised some good points here @danielbate. 

For one thing, I think that users should have the flexibility to execute their contract instances with other wallets and providers. I reckon this can come in handy in many situations and especially during the development stages of a Dapp.

But I do also agree that changing provider and wallet properties might affect the internal state of your Contract instance and might lead to unexpected behaviors or errors.

Having said that, maybe the best approach here would be to try to foresee and handle any errors that could be the outcome of this.

Maybe someone in the team with a deeper understanding of how the SDK handles these situations can help us to better understand this.
